src/frontends/onnx/frontend/src/op/dft.cpp

```diff
@@ -0,0 +1,85 @@
+// Copyright (C) 2018-2023 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "op/dft.hpp"
+
+#include "default_opset.hpp"
+#include "utils/common.hpp"
+
+namespace ngraph {
+namespace onnx_import {
+namespace op {
+namespace set_1 {
+
+namespace {
+// For DFT, IDFT, IRDFT cases, if real data are provided (with shape [D_0, D_1, ..., D_{N-1}, 1])
+// it's needed to fill tensors with zero imaginary part to be aligned with Core ops requirements.
+bool try_convert_real_to_complex(ov::Output<ov::Node>& data) {
+    if (data.get_partial_shape().rank().is_static()) {
+        const auto length = data.get_partial_shape().rank().get_length();
+        const auto last_axis_pos = length - 1;
+        const auto last_dim = data.get_partial_shape()[last_axis_pos];
+        if (last_dim.is_static()) {
+            if (last_dim.get_length() == 1) {
+                ov::Output<ov::Node> imag_part = default_opset::Constant::create(data.get_element_type(), {}, {0});
+                imag_part = std::make_shared<default_opset::Broadcast>(imag_part,
+                                                                       std::make_shared<default_opset::ShapeOf>(data));
+                data = std::make_shared<default_opset::Concat>(OutputVector{data, imag_part}, last_axis_pos);
+                return true;
+            }
+        }
+    }
+    // [D_0, D_1, ..., D_{N-1}, 2] case, so additional transformations not needed or we are not able to check it during
+    // importing.
+    return false;
+}
+}  // namespace
+
+OutputVector dft(const Node& node) {
+    const OutputVector ng_inputs{node.get_ng_inputs()};
+    ov::Output<ov::Node> data = ng_inputs.at(0);
+
+    const auto dft_length_provided = ng_inputs.size() > 1;
```

Comment:
There can be a null node provided as the optional input, it's worth to add a check:
```suggestion
    const auto dft_length_provided = ng_inputs.size() > 1 && !is_null(ng_inputs.at(1));

```

Reply:
a good catch  ;)